lib/model/query/sessions.js

```diff
@@ -14,19 +14,32 @@ const { generateToken, isValidToken } = require('../../util/crypto');
 const { unjoiner } = require('../../util/db');
 const { construct } = require('../../util/util');
 const Option = require('../../util/option');
+const config = require('config');
+const DEFAULT_SESSION_LIFETIME = 24 * 60 * 60;
 
-const aDayFromNow = () => {
-  const date = new Date();
-  date.setDate(date.getDate() + 1);
-  return date;
+const create = (actor, expiresAt) => ({ one }) => {
+  const SESSION_LIFETIME = parseInt(process.env.ODKCENTRAL_SESSION_LIFETIME, 10) || config.sessionLifetime || DEFAULT_SESSION_LIFETIME;
+  const expirySqlFragment = (expiresAt instanceof Date ? sql`${expiresAt.toISOString()}` : sql`statement_timestamp() + ${SESSION_LIFETIME + ' s'}::interval`);
+  return one(sql`
+      INSERT INTO sessions (
+        "actorId",
+        token,
+        csrf,
+        "createdAt",
+        "expiresAt"
+      )
+      VALUES (
+        ${actor.id},
+        ${generateToken()},
+        ${generateToken()},
+        statement_timestamp(),
+        ${expirySqlFragment}
+      )
+      RETURNING *
+    `)
+    .then(construct(Session));
 };
 
-const create = (actor, expiresAt = aDayFromNow()) => ({ one }) => one(sql`
-insert into sessions ("actorId", token, csrf, "createdAt", "expiresAt")
-values (${actor.id}, ${generateToken()}, ${generateToken()}, clock_timestamp(), ${expiresAt.toISOString()})
-returning *`)
-  .then(construct(Session));
```

Comment:
I think the only change here was `statement_timestamp()` to `clock_timestamp()`, but hard to track with the concurrent formatting improvements.